capture/genBitmaps.js

```diff
@@ -169,12 +170,48 @@ function capturePageSelectors(url,scenarios,viewports,bitmaps_reference,bitmaps_
 
           var filePath      = (isReference)?reference_FP:test_FP;
 
-
           if(o === "body:noclip" || o === "document") {
             casper.capture(filePath);
           } else if (casper.exists(o)) {
             if (casper.visible(o)) {
-              casper.captureSelector(filePath, o);
+ 
+                var targetElementsLength,
+                    newTestFiles = [];
+
+                // check if there are more elements on the same page for a given reference selector
+                targetElementsLength = casper.evaluate(function (o) {
+                    var allSimilarElements = document.querySelectorAll(o);
+                    [].forEach.call(allSimilarElements, function(element, index) {
+
+                        // let's add the temporary CSS class name to group the similar elements
+                        element.classList.add('backstopJS' + index); 
+                    });
+                    return allSimilarElements.length;
+                }, o);
+
+                // if there are more elements that has similar CSS seletors
+                if (targetElementsLength > 1) {
+                    var insertPos = fileName.split('_', 2).join('_').length,
+                        testFilesDirectory = filePath.substr(0, filePath.lastIndexOf('/') + 1),
+                        testFileName,
+                        testFilePath,
+                        elementIndex;
+
+                    // take screenshots of all the elements that use the same selector
+                    for (elementIndex = 0; elementIndex < targetElementsLength; elementIndex++) {
+                        testFileName = fileName.substr(0, insertPos) + '_backstopJS' + elementIndex + fileName.substr(insertPos);
```

Comment:
These new filenames are auto generated and users should not get confused with their names, so wanted to let them know explicitly that they are created by backstopJS.

But i can change to '_n' and update the README accordingly.